### ISSUE TITLE
pywincom broken with py2exe?

### ISSUE THREAD

**Issue by nullSoup** [py2exe/py2exe#94]
I'm running into trouble using pywincom with py2exe. It can be reproduced with following minimal example:

test.py:
--------
import sys
import pythoncom

print('Fail?', file=sys.stderr)

--------

setup.py:
--------
from distutils.core import setup
import py2exe

setup(
    windows=['test.py'],
)

--------

Producing this error log when test.exe is run:
--------
Traceback (most recent call last):
  File "test.py", line 2, in <module>
  File "<frozen zipimport>", line 259, in load_module
  File "pythoncom.pyc", line 2, in <module>
ModuleNotFoundError: No module named 'pywintypes'

--------

Python version info:
--------
Python 3.8.10 (tags/v3.8.10:3d8993a, May  3 2021, 11:48:03) [MSC v.1928 64 bit (AMD64)] on win32

--------

Pipfile:
--------
[[source]]
name = "pypi"
url = "https://pypi.org/simple"
verify_ssl = true

[dev-packages]

[packages]
py2exe = "*"
pywin32 = "*"

[requires]
python_version = "3.8"

--------

I noticed that py2exe already includes an explicit hook for pythoncom, so it seems like this probably worked at some point. Running in the debugger, I can confirm that hook_pythoncom() gets called and both pythoncom38.dll and pywintypes38.dll get included in the output dist directory, but apparently pywintypes can't be found at runtime for test.exe.

I originally discovered this issue attempting to use py2exe to package a program using tkfiledialog, which uses this import statement:
from win32com.shell import shell, shellcon

producing this error log:
  File "tkfilebrowser\__init__.pyc", line 21, in <module>
  File "<frozen zipimport>", line 259, in load_module
  File "tkfilebrowser\filebrowser.pyc", line 47, in <module>
  File "<frozen zipimport>", line 259, in load_module
  File "win32com\__init__.pyc", line 6, in <module>
ModuleNotFoundError: No module named 'pythoncom'

Oddly, when I attempt to make a simple test case swapping in that import statement instead, py2exe produces a working distribution, which I don't understand. My one speculation is that there is some issue with the ordering that the imports get processed in as I noticed pythoncom is already getting imported as part of the bootstrap_modules. Maybe something gets loaded after pythoncom while doing the bootstrap_modules loading that destroys the correct setup? And then it does not get loaded again in the main targets section where the actual test.py code is processed because py2exe has marked it as already loaded? I'm probably way off with that speculation but thought I'd throw it out there in case it's a useful thought.

**Comment by jmsadown** [comment]
Thanks for reporting this! I've been scouring the internet for guidance for about a week now.
My version:
Python 3.9.5 (tags/v3.9.5:0a7dcbd, May  3 2021, 17:27:52) [MSC v.1928 64 bit (AMD64)]

**Comment by albertosottile** [comment]
@nullSoup 
> I noticed that py2exe already includes an explicit hook for pythoncom, so it seems like this probably worked at some point. 

That is actually the key of the problem: most of the `pythoncom` code has just been copied from the Python 2.x version of py2exe and never checked with Python 3.x (and so, never maintained in the past 7 years...). 

> Oddly, when I attempt to make a simple test case swapping in that import statement instead, py2exe produces a working distribution, which I don't understand. 

I am not sure I understand this either: do you refer to your test file? Hence, are you telling me that importing `pythoncom` before `sys` fixes the issue for you?

@jmsadown 
> For me, the below imports do not prevent the exe from running as long as I don't actually use any of these modules in the script.
import pywintypes
import pythoncom
import win32com.client

This is also what I would have advised, thank you for this investigation. 

In summary, as you both might have noticed, I do not have the capacity to fully support win32com. However, it appears there are workarounds you both found to make your programs functional. I would encourage you to confirm this/report here your experience so that this issue can be used as reference for the future.
